### PR TITLE
size_t -> int for vec, mat, box etc...

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: Michał Szołucha <mszolucha@nvidia.com>

#### Why we need this PR?
- It fixes a bug with types of ndims in different modules

### CODE REVIEW

**Comment by szalpal** [comment]
!build

File: include/dali/core/geom/vec.h [hunk 1]
```diff
@@ -337,24 +337,24 @@ constexpr auto cross(const vec<2, T> &a, const vec<2, U> &b) {
 }
 
 #define DEFINE_ELEMENTIWSE_VEC_BIN_OP(op)                                              \
-  template <size_t N, typename T, typename U>                                          \
+  template <int N, typename T, typename U>                                          \
```
Comment (klecki) [inline include/dali/core/geom/vec.h]: Please align the macros

File: include/dali/core/geom/vec.h [hunk 2]
```diff
@@ -439,127 +439,127 @@ DALI_HOST_DEV constexpr bool operator!=(const vec<N, T> &a, const vec<N, U> &b)
 ///        with an index `i` ranging from 0 to N-1. `N` must be a compile-time constant
 ///        present at evaluation site.
 #define IMPL_VEC_ELEMENTWISE(...)                                             \
-  size_t i = 0;                                                               \
+  int i = 0;                                                               \
```
Comment (klecki) [inline include/dali/core/geom/vec.h]: And here

File: dali/core/box_test.cc [hunk 3]
```diff
@@ -99,47 +99,47 @@ TEST(BoxSimpleTest, empty_test) {
 
 TEST(BoxSimpleTest, extent_test) {
   ASSERT_EQ(boxes.size(), extents.size()) << "Bad testing data";
-  for (size_t i = 0; i < boxes.size(); i++) {
+  for (int i = 0; i < boxes.size(); i++) {
```
Comment (mzient) [inline dali/core/box_test.cc]: Warning!

File: dali/core/box_test.cc [hunk 4]
```diff
@@ -99,47 +99,47 @@ TEST(BoxSimpleTest, empty_test) {
...
     EXPECT_TRUE(boxes[i].extent() == extents[i]) << "Failed at index: " << i;
   }
 }
 
 
 TEST(BoxSimpleTest, contains_box_test) {
   ASSERT_EQ(boxes.size(), contains_box.size()) << "Bad testing data";
-  for (size_t i = 0; i < boxes.size(); i++) {
+  for (int i = 0; i < boxes.size(); i++) {
```
Comment (mzient) [inline dali/core/box_test.cc]: Warning!

File: dali/core/box_test.cc [hunk 5]
```diff
@@ -99,47 +99,47 @@ TEST(BoxSimpleTest, empty_test) {
...
     EXPECT_EQ(reference_box.contains(boxes[i]), contains_box[i]) << "Failed at index: " << i;
   }
 }
 
 
 TEST(BoxSimpleTest, contains_corner_test) {
   ASSERT_EQ(boxes.size(), contains_corner.size()) << "Bad testing data";
-  for (size_t i = 0; i < boxes.size(); i++) {
+  for (int i = 0; i < boxes.size(); i++) {
```
Comment (mzient) [inline dali/core/box_test.cc]: warning

File: dali/core/box_test.cc [hunk 6]
```diff
@@ -99,47 +99,47 @@ TEST(BoxSimpleTest, empty_test) {
...
     EXPECT_EQ(reference_box.contains(boxes[i].lo), contains_corner[i]) << "Failed at index: " << i;
   }
 }
 
 
 TEST(BoxSimpleTest, overlaps_test) {
   ASSERT_EQ(boxes.size(), overlaps.size()) << "Bad testing data";
-  for (size_t i = 0; i < boxes.size(); i++) {
+  for (int i = 0; i < boxes.size(); i++) {
```
Comment (mzient) [inline dali/core/box_test.cc]: warning

File: dali/core/box_test.cc [hunk 7]
```diff
@@ -99,47 +99,47 @@ TEST(BoxSimpleTest, empty_test) {
...
     EXPECT_EQ(reference_box.overlaps(boxes[i]), overlaps[i]) << "Failed at index: " << i;
   }
 }
 
 
 TEST(BoxSimpleTest, volume_test) {
   ASSERT_EQ(boxes.size(), volumes.size()) << "Bad testing data";
-  for (size_t i = 0; i < boxes.size(); i++) {
+  for (int i = 0; i < boxes.size(); i++) {
```
Comment (mzient) [inline dali/core/box_test.cc]: ...and here...

File: dali/core/box_test.cc [hunk 8]
```diff
@@ -99,47 +99,47 @@ TEST(BoxSimpleTest, empty_test) {
...
     EXPECT_EQ(volume(boxes[i]), volumes[i]) << "Failed at index: " << i;
   }
 }
 
 
 TEST(BoxSimpleTest, intersection_test) {
   ASSERT_EQ(boxes.size(), intersections.size()) << "Bad testing data";
-  for (size_t i = 0; i < boxes.size(); i++) {
+  for (int i = 0; i < boxes.size(); i++) {
```
Comment (mzient) [inline dali/core/box_test.cc]: ...and here.

**Comment by szalpal** [comment]
!build

File: dali/core/box_test.cc (hunk 8 above)
Comment (szalpal) [reply dali/core/box_test.cc]: Done

File: dali/core/box_test.cc (hunk 7 above)
Comment (szalpal) [reply dali/core/box_test.cc]: Done

File: dali/core/box_test.cc (hunk 6 above)
Comment (szalpal) [reply dali/core/box_test.cc]: Done

File: dali/core/box_test.cc (hunk 5 above)
Comment (szalpal) [reply dali/core/box_test.cc]: Done

File: dali/core/box_test.cc (hunk 4 above)
Comment (szalpal) [reply dali/core/box_test.cc]: Done

File: dali/core/box_test.cc (hunk 3 above)
Comment (szalpal) [reply dali/core/box_test.cc]: Done

File: include/dali/core/geom/vec.h (hunk 2 above)
Comment (szalpal) [reply include/dali/core/geom/vec.h]: Done

File: include/dali/core/geom/vec.h (hunk 1 above)
Comment (szalpal) [reply include/dali/core/geom/vec.h]: Done

File: include/dali/core/geom/mat.h [hunk 9]
```diff
@@ -297,24 +297,24 @@ constexpr bool operator==(const mat<rows, cols, T> &a, const mat<rows, cols, U>
...
-    for (size_t i = 0; i < sub_r; i++)
+  for (int j = 0; j < sub_c; j++)
+    for (int i = 0; i < sub_r; i++)
       result(i, j) = m(i+r, j+c);
   return result;
 }
 
 #define DEFINE_ELEMENTWISE_MAT_MAT_BINOP(op)                            \
-  template <size_t rows, size_t cols, typename T1, typename T2>         \
+  template <int rows, int cols, typename T1, typename T2>         \
```
Comment (mzient) [inline include/dali/core/geom/mat.h]: formatting

File: include/dali/core/geom/mat.h [hunk 10]
```diff
@@ -325,7 +325,7 @@ constexpr auto sub(const mat<rows, cols, Element> &m, int r, int c) {
   }
 
 #define DEFINE_ELEMENTWISE_RHS_BINOP(op)                            \
-  template <size_t rows, size_t cols, typename T1, typename T2,     \
+  template <int rows, int cols, typename T1, typename T2,     \
```
Comment (mzient) [inline include/dali/core/geom/mat.h]: formatting

File: include/dali/core/geom/mat.h [hunk 11]
```diff
@@ -337,7 +337,7 @@ constexpr auto sub(const mat<rows, cols, Element> &m, int r, int c) {
   }
 
 #define DEFINE_ELEMENTWISE_LHS_BINOP(op)                            \
-  template <size_t rows, size_t cols, typename T1, typename T2,     \
+  template <int rows, int cols, typename T1, typename T2,     \
```
Comment (mzient) [inline include/dali/core/geom/mat.h]: formatting

File: include/dali/core/geom/vec.h [hunk 12]
```diff
@@ -285,12 +285,12 @@ struct vec : vec_base<N, T> {
 #define DEFINE_ASSIGN_VEC_OP(op)                                                         \
   template <typename U>                                                                  \
   DALI_HOST_DEV vec &operator op(const vec<N, U> &rhs) {                                 \
-    for (size_t i = 0; i < N; i++) v[i] op rhs[i];                                       \
+    for (int i = 0; i < N; i++) v[i] op rhs[i];                                       \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

File: include/dali/core/geom/vec.h [hunk 13]
```diff
@@ -285,12 +285,12 @@ struct vec : vec_base<N, T> {
...
     return *this;                                                                        \
   }                                                                                      \
   template <typename U>                                                                  \
   DALI_HOST_DEV std::enable_if_t<is_scalar<U>::value, vec &> operator op(const U &rhs) { \
-    for (size_t i = 0; i < N; i++) v[i] op rhs;                                          \
+    for (int i = 0; i < N; i++) v[i] op rhs;                                          \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

File: include/dali/core/geom/vec.h [hunk 14]
```diff
@@ -372,24 +372,24 @@ DEFINE_ELEMENTIWSE_VEC_BIN_OP(<=)
 DEFINE_ELEMENTIWSE_VEC_BIN_OP(>=)
 
 #define DEFINE_SHIFT_VEC_BIN_OP(op)                                                                \
-  template <size_t N, typename T, typename U>                                                      \
+  template <int N, typename T, typename U>                                                      \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

File: include/dali/core/geom/vec.h [hunk 15]
```diff
@@ -372,24 +372,24 @@ DEFINE_ELEMENTIWSE_VEC_BIN_OP(<=)
...
   DALI_HOST_DEV vec<N, T> operator op(const vec<N, T> &a, const vec<N, U> &b) {                    \
     vec<N, T> ret;                                                                                 \
-    for (size_t i = 0; i < N; i++) ret[i] = a[i] op b[i];                                          \
+    for (int i = 0; i < N; i++) ret[i] = a[i] op b[i];                                          \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

File: include/dali/core/geom/vec.h [hunk 16]
```diff
@@ -372,24 +372,24 @@ DEFINE_ELEMENTIWSE_VEC_BIN_OP(<=)
...
     return ret;                                                                                    \
   }                                                                                                \
-  template <size_t N, typename T, typename U>                                                      \
+  template <int N, typename T, typename U>                                                      \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

File: include/dali/core/geom/vec.h [hunk 17]
```diff
@@ -372,24 +372,24 @@ DEFINE_ELEMENTIWSE_VEC_BIN_OP(<=)
...
   DALI_HOST_DEV std::enable_if_t<is_scalar<U>::value, vec<N, T>> operator op(const vec<N, T> &a,   \
                                                                              const U &b) {         \
     vec<N, T> ret;                                                                                 \
-    for (size_t i = 0; i < N; i++) ret[i] = a[i] op b;                                             \
+    for (int i = 0; i < N; i++) ret[i] = a[i] op b;                                             \
     return ret;                                                                                    \
   }                                                                                                \
-  template <size_t N, typename T, typename U>                                                      \
+  template <int N, typename T, typename U>                                                      \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

File: include/dali/core/geom/vec.h [hunk 18]
```diff
@@ -372,24 +372,24 @@ DEFINE_ELEMENTIWSE_VEC_BIN_OP(<=)
...
+    for (int i = 0; i < N; i++) ret[i] = a[i] op b[i];                                          \
     return ret;                                                                                    \
   }                                                                                                \
-  template <size_t N, typename T, typename U>                                                      \
+  template <int N, typename T, typename U>                                                      \
   DALI_HOST_DEV std::enable_if_t<is_scalar<U>::value, vec<N, T>> operator op(const vec<N, T> &a,   \
                                                                              const U &b) {         \
     vec<N, T> ret;                                                                                 \
-    for (size_t i = 0; i < N; i++) ret[i] = a[i] op b;                                             \
+    for (int i = 0; i < N; i++) ret[i] = a[i] op b;                                             \
```
Comment (mzient) [inline include/dali/core/geom/vec.h]: formatting

**Pull request merged**